src/OpenTelemetry.Resources.Host/HostDetector.cs

```diff
@@ -94,6 +94,14 @@ public Resource Detect()
             {
                 attributes.Add(new(HostSemanticConventions.AttributeHostId, machineId));
             }
+#if !NETFRAMEWORK
+            // Architecture is only supported in .NET 5+
+            var arch = RuntimeInformation.ProcessArchitecture.ToString();
```

Comment:
I think that you need to use some custom mapping for the expected values. From: https://github.com/open-telemetry/semantic-conventions/blob/v1.37.0/docs/resource/host.md

host.arch has the following list of well-known values. If one of them applies, then the respective value MUST be used; otherwise, a custom value MAY be used.

For the values not defined in the table, probbly ToLowerInvariant() will be better option.